src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/LinearSpaceBlock.java

```diff
@@ -0,0 +1,89 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.broadinstitute.hdf5.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * This class represents a key for a contiguous block of elements in an indexed linear space.
+ *
+ * The begin index is include
+ * The end index is exclusive
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class LinearSpaceBlock implements Serializable {
+
+    private static final long serialVersionUID = 4571138983754570342L;
+
+    /* begin index inclusive, end index exclusive */
+    private final int begIndex, endIndex, numTargets;
+
+    public LinearSpaceBlock(final int begIndex, final int endIndex) {
+        this.begIndex = ParamUtils.isPositiveOrZero(begIndex, "The begin index of a block must be non-negative.");
+        this.endIndex = ParamUtils.inRange(endIndex, begIndex + 1, Integer.MAX_VALUE, "The block must at least" +
```

Comment:
Used `Utils.validateArg(endIndex > begIndex, "The block must contain at least one element");` in the beginning instead.